packages/router/src/__tests__/util.test.ts

```diff
@@ -307,9 +309,9 @@ describe('replaceParams', () => {
       '/boolean/false'
     )
 
-    expect(replaceParams('/undef/{undef}', { undef: undefined })).toEqual(
-      '/undef/{undef}'
-    )
+    expect(() =>
+      replaceParams('/undef/{undef}', { undef: undefined })
+    ).toThrowError("Missing parameter 'undef' for route '/undef/{undef}'.")
```

Comment:
This still sticks out like a sore thumb to me 😃
If I ran in to this as a user I'd be like "It's not missing. It's right there! I set it to `undefined`!"

But this is not what this PR is about, so let's ignore that 🙂 
Besides, similar to your argumentation for this PR - let's leave it as it is, and when/if someone complains about it or raises an issue about it we can take another look.